### PR TITLE
Add Swedish `mobileNumber()`

### PR DESCRIPTION
**Pull request by tanthammar**
Add new feature: Swedish mobileNumber() format

New documentation is PR:ed  https://github.com/FakerPHP/fakerphp.github.io/pull/71

### CODE REVIEW

**Comment by pimjansen** [comment]
Can you add some references in the formats where we can verify the structure of the formats and for future reference?

**Comment by tanthammar** [comment]
I tried to find references online. Found only for Swedish landline. So I added the formats I know, with the two most common patterns for Mobile phone numbers in Sweden. 
I followed the structure in the Norwegian equivalent method.

**Comment by tanthammar** [comment]
I mocked 100 numbers and ran them all through Brick mobile number validator. They all passed.

**Comment by pimjansen** [comment]
I do however need some reference to verify the formats. The PR i also missing unittests for the new data

**Comment by tanthammar** [comment]
I've asked for assistance in a Swedish slack channel. Hopefully someone can assist with a reference. Will fix the unit tests.

**Comment by tanthammar** [comment]
Here is an official reference https://www.pts.se/sv/bransch/telefoni/nummer-och-adressering/telefoninummerplanen/telefonnummers-struktur/

**Comment by pimjansen** [comment]
> Here is an official reference https://www.pts.se/sv/bransch/telefoni/nummer-och-adressering/telefoninummerplanen/telefonnummers-struktur/

Can we add that to the phpdoc with an @see?

**Comment by tanthammar** [comment]
@pimjansen How do I reopen, I've added the @see to phpdoc.

**Comment by tanthammar** [comment]
Do I need to change anything or is all ok?

**Comment by tanthammar** [comment]
@bram-pkg Can you help me please. Don't see what is wrong.

**Comment by bram-pkg** [comment]
It seems your phpdoc is not aligned properly, you can check out the check's output

**Comment by localheinz** [comment]
@tanthammar 

Run

```shell
make cs
```

to fix coding standard issues.

**Comment by localheinz** [comment]
@Nyholm 

Can you verify that this makes sense?

@pimjansen 

Not sure where we are with adding features at the moment, are we still up for it?

File: src/Faker/Provider/sv_SE/PhoneNumber.php [hunk 1]
```diff
@@ -34,4 +37,24 @@ class PhoneNumber extends \Faker\Provider\PhoneNumber
...
+        '+467########',
+        '+46(0)7########',
+        '+46 (0)7## ## ## ##',
+        '+46 (0)7## ### ###',
+        '07## ## ## ##',
+        '07## ### ###',
+
+    ];
+
+    public function mobileNumber()
```
Comment (localheinz) [inline src/Faker/Provider/sv_SE/PhoneNumber.php:54]:
```suggestion
    public function mobileNumber(): string
```

File: src/Faker/Provider/sv_SE/PhoneNumber.php [hunk 2]
```diff
@@ -34,4 +37,24 @@ class PhoneNumber extends \Faker\Provider\PhoneNumber
...
+     */
+    protected static $mobileFormats = [
+        '+467########',
+        '+46(0)7########',
+        '+46 (0)7## ## ## ##',
+        '+46 (0)7## ### ###',
+        '07## ## ## ##',
+        '07## ### ###',
+
+    ];
```
Comment (localheinz) [inline src/Faker/Provider/sv_SE/PhoneNumber.php:52]:
```suggestion
        '07## ### ###',
    ];
```

File: src/Faker/Provider/sv_SE/PhoneNumber.php [hunk 3]
```diff
@@ -34,4 +37,24 @@ class PhoneNumber extends \Faker\Provider\PhoneNumber
         '+46(0)%######',
         '+46%######',
     ];
+
+    /**
+     * @var array Swedish mobile number formats
+     */
+    protected static $mobileFormats = [
```
Comment (localheinz) [inline src/Faker/Provider/sv_SE/PhoneNumber.php:44]:
```suggestion
    protected static array $mobileFormats = [
```

File: src/Faker/Provider/sv_SE/PhoneNumber.php [hunk 4]
```diff
@@ -34,4 +37,24 @@ class PhoneNumber extends \Faker\Provider\PhoneNumber
         '+46(0)%######',
         '+46%######',
     ];
+
+    /**
+     * @var array Swedish mobile number formats
```
Comment (localheinz) [inline src/Faker/Provider/sv_SE/PhoneNumber.php:42]:
```suggestion
     * @var array<int, string> Swedish mobile number formats
```

**Comment by localheinz** [comment]
@tanthammar 

Are you up for adding a test?

**Review by localheinz — COMMENTED** [review]
Apart from lacking tests, this looks good to me.

**Comment by tanthammar** [comment]
I can try to replicate a test from nb_NO/PhoneNumberTest.php, if that covers the requirements?

**Comment by tanthammar** [comment]
OK, did a test.

File: src/Faker/Provider/sv_SE/PhoneNumber.php [hunk 5]
```diff
@@ -34,4 +37,23 @@ class PhoneNumber extends \Faker\Provider\PhoneNumber
...
+    /**
+     * @var array<int, string> Swedish mobile number formats
+     */
+    protected static array $mobileFormats = [
+        '+467########',
+        '+46(0)7########',
+        '+46 (0)7## ## ## ##',
+        '+46 (0)7## ### ###',
+        '07## ## ## ##',
+        '07## ### ###',
```
Comment (Nyholm) [inline src/Faker/Provider/sv_SE/PhoneNumber.php:50]: Thank you for this PR. 
Also add: 

```
07##-## ## ##
07##-### ###
07# ### ## ##
07#-### ## ##
07#-#######
```

etc

**Comment by localheinz** [comment]
@tanthammar 

Can you rebase on top of `main`, please?

File: src/Faker/Provider/sv_SE/PhoneNumber.php (hunk 5 above)
Comment (tanthammar) [reply src/Faker/Provider/sv_SE/PhoneNumber.php:50]: Added these patterns and altered the test.

**Comment by tanthammar** [comment]
@localheinz 

rebased, main :)

File: test/Faker/Provider/sv_SE/PhoneNumberTest.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * @group legacy
+ */
+final class PhoneNumberTest extends TestCase
+{
+    public function testMobileNumber(): void
+    {
+        for ($i = 0; $i < 10; ++$i) {
+            $number = $this->faker->mobileNumber;
```
Comment (localheinz) [inline test/Faker/Provider/sv_SE/PhoneNumberTest.php:16]:
```suggestion
            $number = $this->faker→mobileNumber();
```

**Comment by tanthammar** [comment]
Ouch, don't remember string_starts_with() from php 7.4... 

Do you? for the test...

**Comment by localheinz** [comment]
@tanthammar 

`str_starts_with()` was added with PHP 8.0, see https://www.php.net/manual/en/migration80.new-features.php#migration80.new-features.standard.

**Comment by tanthammar** [comment]
Changed to strpos() in the test to please php@7.4

**Review by bram-pkg — APPROVED** [review]

**Comment by bram-pkg** [comment]
Thank you @tanthammar :)

And sorry for the slow correspondence!

**Comment by Nyholm** [comment]
Wohoo. 

Well done!

**Pull request merged**